lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -212,6 +224,24 @@ def chefignore
         @chefignore ||= Chefignore.new(File.basename(cookbook_path))
       end
 
+      def load_all_unignored_files
+        # List all directories that are not dotdirs, then list everything in
+        # those, *including* dotfiles and dotdirs.
+        # Finally, list all the _files_ at the root, including dotfiles, and
+        # merge those in.
```

Comment:
I've been mulling this over, and wonder if this weird exception is really necessary. On the plus side, it prevents you from enumerating a `.git` or `.svn`, which is nice. Also, if you don't have a `chefignore` you still get your `.git` dir ignored for free and I think vagrant+VMWare fusion puts stuff in a dotdir also, which would be nice to auto-ignore. OTOH, if you actually wanted to keep a dotdir in the root, there would be nothing you could do. Not sure what that use case would be though; I can't think of anything.